samples/cpp/text2image/txt2image_from_subcomponent.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "imwrite.hpp"
+#include "openvino/genai/image_generation/text2image_pipeline.hpp"
+
+int32_t main(int32_t argc, char* argv[]) try {
+    OPENVINO_ASSERT(argc >= 3 && argc <= 6,
+                    "Usage: ",
+                    argv[0],
+                    " <MODEL_DIR> '<PROMPT>' [ <TXT_ENCODE_DEVICE> <UNET_DEVICE> <VAE_DEVICE> ]");
+
+    const std::string models_path = argv[1], prompt = argv[2];
+
+    std::filesystem::path root_dir = models_path;
+
+    const int width = 512;
+    const int height = 512;
+    const float guidance_scale = 7.5f;
+    const int number_of_images_to_generate = 1;
+    const int number_of_inference_steps_per_image = 20;
+
+    // Set devices to command-line args if specified, otherwise default to CPU.
+    // Note that these can be set to CPU, GPU, or NPU.
+    const std::string text_encoder_device = (argc > 3) ? argv[3] : "CPU";
+    const std::string unet_device = (argc > 4) ? argv[4] : "CPU";
+    const std::string vae_decoder_device = (argc > 5) ? argv[5] : "CPU";
+
+    std::cout << "text_encoder_device: " << text_encoder_device << std::endl;
+    std::cout << "unet_device: " << unet_device << std::endl;
+    std::cout << "vae_decoder_device: " << vae_decoder_device << std::endl;
+
+    // this is the path to where compiled models will get cached
+    // (so that the 'compile' method run much faster 2nd+ time)
+    std::string ov_cache_dir = "./cache";
+
+    //
+    // Step 1: Prepare each Text2Image subcomponent (scheduler, text encoder, unet, vae) separately.
+    //
+
+    // Create the scheduler from the details listed in the json.
+    auto scheduler = ov::genai::Scheduler::from_config(root_dir / "scheduler/scheduler_config.json");
+
+    // Note that we could have created the scheduler by specifying specific type (for example EULER_DISCRETE), like
```

Comment:
we can highlight that it can be useful when Scheduler is not supported natively by OpenVINO GenAI (but we are working to support most commonly used ones)